python/cuml/cuml/tests/conftest.py

```diff
@@ -253,45 +383,61 @@ def nlp_20news():
 
 @pytest.fixture(scope="session")
 def housing_dataset():
-    try:
-        data = fetch_california_housing()
+    """Load and preprocess the California housing dataset.
+
+    This fixture loads the California housing dataset and returns the
+    feature matrix, target vector, and feature names.
+
+    Returns
+    -------
+    tuple
+        (X, y, feature_names) where X is the feature matrix, y is the target
+        vector, and feature_names is a list of feature names
+    """
+
+    @dataset_fetch_retry
+    def _fetch_housing():
+        return fetch_california_housing()
 
-    # failing to download has appeared as multiple varied errors in CI
-    except:  # noqa E722
-        pytest.xfail(reason="Error fetching housing dataset")
+    try:
+        data = _fetch_housing()
+    except Exception as e:
+        pytest.xfail(f"Error fetching housing dataset: {str(e)}")
```

Comment:
I recognize this is already done from before, but couldn't find a discussion of this practice.

xfailing a test that fails to download a dataset won't show up as a clear signal - our test logs are very long and since this won't fail the test people will have no reason to check if the download failed.

I'd rather see us move away from downloading datasets that we might expect to fail. If we need large datasets we can generate them locally, and small datasets should download fine or could be included in the repo if needed. If we do need to download a dataset for a test, having it fail to download should fail the test IMO.

Reply:
Alternatively, we could gate these tests behind a `--uses-network` flag added to pytest. By default then these tests wouldn't run (say for long term support builds), but on CI we'd add this flag to always run them.

Reply:
I generally agree with your arguments. I'd suggest we leave that for a follow-up since the current PR would not fundamentally change the current behavior.

Reply:
Oh, definitely, just wanted to note it somewhere. I'll open a new issue.